tt_metal/impl/program/program.hpp

```diff
@@ -101,6 +101,12 @@ struct ProgramConfig {
 
 inline namespace v0 {
 
+enum class ProgramBinaryStatus : uint8_t {
```

Comment:
Would be great to explain what each of these mean here, and the lifecycle of the status changes.

Reply:
I've added a description for each field and the enum itself.